nat-lab/tests/test_teliod.py

```diff
@@ -45,41 +52,21 @@ async def test_teliod() -> None:
         assert (
             "Command executed successfully"
             == (
-                await asyncio.wait_for(
-                    connection.create_process(TELIOD_HELLO_WORLD_PARAMS).execute(),
-                    1,
-                )
+                await connection.create_process(TELIOD_HELLO_WORLD_PARAMS).execute()
             ).get_stdout()
         )
 
         assert teliod_process.is_executing()
 
-        # Get Teliod PID
-        teliod_pid = (
-            await asyncio.wait_for(
-                connection.create_process(["pidof", "teliod"]).execute(),
-                1,
-            )
-        ).get_stdout()
-
-        print(f"Teliod PID: {teliod_pid}")
-
         # Send SIGTERM to the daemon
-        await asyncio.wait_for(
-            connection.create_process(["kill", f"{teliod_pid}"]).execute(),
-            1,
-        )
-
-        # Let the daemon stop
-        await asyncio.sleep(1)
+        await connection.create_process(
+            ["killall", "-w", "-s", "SIGTERM", "teliod"]
+        ).execute()
 
         assert not teliod_process.is_executing()
 
         # Run the hello-world command again - this time it should fail
         try:
-            await asyncio.wait_for(
-                connection.create_process(TELIOD_HELLO_WORLD_PARAMS).execute(),
-                1,
-            )
+            await connection.create_process(TELIOD_HELLO_WORLD_PARAMS).execute()
```

Comment:
You forgot `assert False` in the try so if the execute does not fail you won't know about it. That's why it's probably better to use pytest.raises like this:
```
with pytest.raises(ProcessExecError) as err:
    await connection.create_process(TELIOD_HELLO_WORLD_PARAMS).execute()
assert err.stderr == "Error: DaemonIsNotRunning"
```